central/image/datastore/store/common/v2/merge_v2.go

```diff
@@ -0,0 +1,68 @@
+package common
+
+import (
+	"github.com/stackrox/rox/central/cve/converter/utils"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+// MergeV2 merges the images parts into an image.
+func MergeV2(parts ImageParts) *storage.Image {
+	mergeComponentsV2(parts, parts.Image)
+	return parts.Image
+}
+
+func mergeComponentsV2(parts ImageParts, image *storage.Image) {
+	// If the image has a nil scan, there is nothing to fill in.
+	if image.GetScan() == nil {
+		return
+	}
+
+	for _, cp := range parts.Children {
+		if cp.ComponentV2 == nil {
+			log.Errorf("UNEXPECTED: nil component when retrieving components for image %q", image.GetId())
+			continue
+		}
+		// Generate an embedded component from the non-embedded version.
+		image.Scan.Components = append(image.Scan.Components, generateEmbeddedComponentV2(cp))
+	}
+}
+
+func generateEmbeddedComponentV2(cp ComponentParts) *storage.EmbeddedImageScanComponent {
+	vulns := make([]*storage.EmbeddedVulnerability, 0, len(cp.Children))
+	for _, cve := range cp.Children {
+		if cve.CVEV2 == nil {
+			log.Errorf("UNEXPECTED: nil cve when retrieving cves for component %q", cp.Component.GetId())
+			continue
+		}
+		vulns = append(vulns, generateEmbeddedCVEV2(cve.CVEV2))
+	}
+
+	ret := &storage.EmbeddedImageScanComponent{
+		Name:         cp.ComponentV2.GetName(),
+		Version:      cp.ComponentV2.GetVersion(),
+		License:      cp.ComponentV2.GetLicense().CloneVT(),
```

Comment:
Do we even use this? Where is this ever even populated?

Reply:
First glance, doesn't look to be used.  I wrote https://issues.redhat.com/browse/ROX-28186 to defer and clean it up later to minimize size of this PR.